snuba/settings.py

```diff
@@ -69,9 +69,23 @@
 BULK_CLICKHOUSE_BUFFER = 10000
 
 # Processor/Writer Options
-DEFAULT_BROKERS = ["localhost:9092"]
+DEFAULT_BROKERS: Sequence[str] = os.environ.get(
+    "DEFAULT_BROKERS", "localhost:9092"
+).split(",")
 DEFAULT_STORAGE_BROKERS: Mapping[str, Sequence[str]] = {}
 
+_BROKER_CONFIG: Mapping[str, Any] = {
+    "bootstrap.servers": ",".join(DEFAULT_BROKERS),
+    "sasl.mechanism": os.environ.get("SASL_MECHANISM"),
+    "sasl.username": os.environ.get("SASL_PLAIN_USERNAME"),
+    "sasl.password": os.environ.get("SASL_PLAIN_PASSWORD"),
+    "security.protocol": os.environ.get("SECURITY_PROTOCOL"),
+}
+BROKER_CONFIG: Mapping[str, Any] = {
+    k: v for k, v in _BROKER_CONFIG.items() if v is not None
```

Comment:
I will now do the function that loads the config, which strips out the None value, and remove this.

Reply:
Thanks. 
Could you please also validate in that function that the user is specifying parameters we support? This was the intention with the dataclass, but now we are not doing it anymore. A list of allowed parameters in the function is more than enough and you can start by filling it in with the ones you need. 
This would allow us to have a contract to what users may be overriding when we make changes to the config structure and still keep it very easy to support new ones. Making it more restrictive later for any reason would instead be very hard.
(raising an error if an unsupported parameter is found would be desirable).